File: code-examples/pymongo/authentication/cr.py
from pymongo import MongoClient
import urllib.parse

# Replace the following with values for your environment.
username = urllib.parse.quote_plus('<username>')
password = urllib.parse.quote_plus('<password>')
clusterUrl = '<MongoDB cluster url>'

# Replace the following with your MongoDB deployment's connection string.
uri = f"mongodb+srv://{username}:{password}@{clusterUrl}/?authMechanism=SCRAM-SHA-256&tls=true&tlsCertificateKeyFile=<clientPEMFile>"

# Create a new MongoClient
client = MongoClient(uri)

# Function to connect to the server
def run():
    try:
        # Establish and verify connection
        client["admin"].command("ping")
        print("Connected successfully to server")
    except Exception as e:
        print(e)
    finally:
        # Ensures that the client will close when you finish/error
        client.close()

run()

File: code-examples/c/authentication/cr.c
#include <mongoc.h>
#include <bson.h>
#include <stdio.h>

int main (int argc, char **argv)
{
   mongoc_client_t *client;
   bson_error_t error;
   char *str;
   bson_t command, reply;
   bson_t *opts;
   
   // Initialize the MongoDB C driver.
   mongoc_init ();

   // Replace the following with values from your environment.
   const char *mongourl = "mongodb+srv://<username>:<password>@<MongoDB cluster url>/?authMechanism=SCRAM-SHA-256&tls=true&tlsCertificateKeyFile=<path to PEM file>";

   // Create a new MongoDB client.
   client = mongoc_client_new (mongourl);
   if (!client) {
      fprintf (stderr, "Failed to parse URI.\n");
      return EXIT_FAILURE;
   }
   
   // Establish connection and verify connection
   bson_init (&command);
   bson_append_int32 (&command, "ping", 4, 1);
   
   // Run the 'ping' command to verify connection
   if (!mongoc_client_command_simple (client, "admin", &command, NULL, &reply, &error)) {
      fprintf (stderr, "Ping failed: %s\n", error.message);
   } else {
       // Print successful connection message
       printf ("Connected successfully to server\n");
   }

   str = bson_as_json (&reply, NULL);
   
   // Clean up retrieved data
   bson_free (str);
   bson_destroy (&reply);
   bson_destroy (&command);

   // Ensures that the client will close when you finish/error
   mongoc_client_destroy (client);

   // Release resources.
   mongoc_cleanup ();
   
   return EXIT_SUCCESS;
}
